### ISSUE TITLE
Pick up internal import files

### ISSUE THREAD

**Issue by jkransen** [castlemock/castlemock#179]
In a docker context, it is convenient to be able to add a file to an "imports" subdirectory that is accessed by castlemock. Upon startup, castlemock can import these files. This bypasses the need to use the REST API, and the accompanied timing issues between creating the docker image and accessing the actually running container.

**Describe the solution you'd like**
A subdirectory called something like "imports",  that is scanned periodically, but at least upon startup. Any files in it are directly imported, and possibly deleted. Alternative to deletion is just to ignore any projects with id that already exists. 

**Describe alternatives you've considered**
Other ways to create a docker image with an imported project seem to be harder. The REST API for importing is only available in a running container, and not upon image creation.

**Additional context**
None.

**Comment by GrumpyMeow** [comment]
I agree with this request. It’s something I was also trying to resolve. Shouldn’t be too hard to implement to scan for a file upon startup.

**Comment by karldahlgren** [comment]
Hi @jkransen.
I absolutely agree that this would be a nice feature! I will look into it as soon as possible.
However, this you manage to see what @GrumpyMeow posted before regarding the persistence outside the Docker image? Is that something that would help for now? 

Wiki: https://github.com/castlemock/castlemock/wiki/Docker#persistence

Best regards,
Karl Dahlgren

**Comment by jkransen** [comment]
Hi @karldahlgren , yes I did consider this, but when I want to distribute a Docker image that mocks a real world endpoint, then the existing solution would require distributing the project files along with the image. The proposed feature would allow only distributing the newly created image. It's good to have both options available :-) Can you give me a heads up once you actually want to start on it? I may give it a try myself, and it would be a waste if we did things in parallel. :-)

**Comment by GrumpyMeow** [comment]
Hi Jeroen,
Are you aware that it’s also possible to create a Docker image ontop of the dockerhub image?
In your own Dockerfile you can then provision the files in the folder that Castlemock uses for internal storage.
You’ll than not use the import/export xml any more. But use a whole set of small xml files. I’ve committed these files to a git repository. This seems to work quite good.

**Comment by jkransen** [comment]
Hi @GrumpyMeow , I had thought about docker volume and adding the files through Dockerfile, but not to combine them .That's actually a good idea, having an "edit mode" with a docker volume and a "persist" mode with copying the files through Dockerfile. Glad that this is possible. To be honest, I was not totally sure if the .castlemock files would fully cover the state, glad to know that it does.

**Comment by GrumpyMeow** [comment]
Yes, the files in the .castlemock folder cover the full state. I did put some .gitignore entries in place to exclude a few file types. For example: “**/*.user” and “**/*.token”.
If you’re gonna resolve this issue in Docker, it might be interesting to look at the “ONBUILD” Dockerfile-instruction for this purpose. This instruction was new for me and peaked my interest, but i haven’t got around to trying it out yet.

Veel succes!

**Comment by jkransen** [comment]
Hi @GrumpyMeow , I can now run the "empty" castlemock docker image, save the files locally under 'cmfiles' in a git repo. I use this Dockerfile to create a "filled" castlemock image:

`FROM castlemock/castlemock:latest

COPY cmfiles/* /root/.castlemock/
`

When I build this docker image, and run it, I see an empty project list, but al the files are there when I /bin/bash into the container. What am I missing?
